### PR TITLE
Issue #3482526: Add redirect from /edit-profile to edit profile page of current user

### PR DESCRIPTION
**Pull request by vcsvinicius**
## Problem (for internal)
Add a redirect from /edit-profile path to /user/{user-id}/profile (edit profile page) of current user. It will smoth the user-experience.

## Solution (for internal)
Create a route for logged user and add a redirect to edit profile page of logged user.

## Release notes (to customers)
Add route for edit-profile to improve the user-experience

## Issue tracker
[PROD-30751](https://getopensocial.atlassian.net/browse/PROD-30751)
[#3482526](https://www.drupal.org/project/social/issues/3482526)

## Theme issue tracker
N/A

## How to test
- [ ] Access with logged user
- [ ] Go to /edit-profile and should be redirect to edit profile page

## Change Record
N/A

## Translations
N/A

[PROD-30751]: https://getopensocial.atlassian.net/browse/PROD-30751?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Comment by open-social-tugboat** [comment]
[Tugboat](https://www.tugboatqa.com) has finished building the preview for this pull request!

Link:
- https://pr4134-0tzpv5k2we7enrbv7adlld0e2udh7o0i.tugboatqa.com


Dashboard:
- https://dashboard.tugboatqa.com/6717d2d2e4d2024934a35bff

**Comment by open-social-tugboat** [comment]
[Tugboat](https://www.tugboatqa.com) has finished building the preview for this pull request!

Link:
- https://pr4134-0tzpv5k2we7enrbv7adlld0e2udh7o0i.tugboatqa.com


Dashboard:
- https://dashboard.tugboatqa.com/6717d2d2e4d2024934a35bff

**Comment by open-social-tugboat** [comment]
[Tugboat](https://www.tugboatqa.com) has finished building the preview for this pull request!

Link:
- https://pr4134-0tzpv5k2we7enrbv7adlld0e2udh7o0i.tugboatqa.com


Dashboard:
- https://dashboard.tugboatqa.com/6717d2d2e4d2024934a35bff

File: tests/behat/features/capabilities/account/edit-profile-page.feature [hunk 1]
```diff
@@ -0,0 +1,12 @@
+@api
```
Comment (denis-getopensocial) [inline tests/behat/features/capabilities/account/edit-profile-page.feature:1]: I believe this page could have been name edit-profile-redirection.
What do you think?

File: tests/behat/features/capabilities/account/edit-profile-page.feature [hunk 2]
```diff
@@ -0,0 +1,12 @@
...
+  Benefit: To make it easier for users to access their edit profile page
+  Role: As a Authenticated
+  Goal/desire: I want to be redirected to "/user/*/profile" when I go to "/edit-profile"
+
+  Scenario: A user is redirected to their edit profile page
+    Given I am logged in as a user with the "authenticated" role
+
+    When I go to "/edit-profile"
+
+    Then the URL should match "^\/user\/[^\/]+\/profile"
```
Comment (denis-getopensocial) [inline tests/behat/features/capabilities/account/edit-profile-page.feature:12]: By doing the same myself I believe URL should be in lower case here no?

File: modules/social_features/social_profile/src/Form/SocialProfileSettingsForm.php [hunk 3]
```diff
@@ -193,4 +194,17 @@ public function submitForm(array &$form, FormStateInterface $form_state) {
     parent::submitForm($form, $form_state);
   }
 
+  /**
+   * Redirects users from /edit-profile to edit profile page of current user.
+   *
+   * @return \Symfony\Component\HttpFoundation\RedirectResponse
+   *   Returns a redirect to the edit profile page of current user.
+   */
+  public function editProfileRedirect(): RedirectResponse {
```
Comment (denis-getopensocial) [inline modules/social_features/social_profile/src/Form/SocialProfileSettingsForm.php:203]: This needs to be its own controller, or at least based on a class that extend ControllerBase.

**Review by denis-getopensocial — CHANGES_REQUESTED** [review]

File: tests/behat/features/capabilities/account/edit-profile-page.feature (hunk 1 above)
Comment (vcsvinicius) [reply tests/behat/features/capabilities/account/edit-profile-page.feature:1]: I follow the pattern used in https://github.com/goalgorilla/open_social/pull/4097
So if needs be changed here, I guess should be better change there as well.

**Comment by open-social-tugboat** [comment]
[Tugboat](https://www.tugboatqa.com) has finished building the preview for this pull request!

Link:
- https://pr4134-0tzpv5k2we7enrbv7adlld0e2udh7o0i.tugboatqa.com


Dashboard:
- https://dashboard.tugboatqa.com/6717d2d2e4d2024934a35bff

File: tests/behat/features/capabilities/account/edit-profile-page.feature (hunk 1 above)
Comment (denis-getopensocial) [reply tests/behat/features/capabilities/account/edit-profile-page.feature:1]: I have to admit i didnt look at the file name there - I guess yes you can update the file name part of this PR

**Comment by open-social-tugboat** [comment]
[Tugboat](https://www.tugboatqa.com) has finished building the preview for this pull request!

Link:
- https://pr4134-0tzpv5k2we7enrbv7adlld0e2udh7o0i.tugboatqa.com


Dashboard:
- https://dashboard.tugboatqa.com/6717d2d2e4d2024934a35bff

**Comment by open-social-tugboat** [comment]
[Tugboat](https://www.tugboatqa.com) has finished building the preview for this pull request!

Link:
- https://pr4134-0tzpv5k2we7enrbv7adlld0e2udh7o0i.tugboatqa.com


Dashboard:
- https://dashboard.tugboatqa.com/6717d2d2e4d2024934a35bff

**Comment by open-social-tugboat** [comment]
[Tugboat](https://www.tugboatqa.com) has finished building the preview for this pull request!

Link:
- https://pr4134-0tzpv5k2we7enrbv7adlld0e2udh7o0i.tugboatqa.com


Dashboard:
- https://dashboard.tugboatqa.com/6717d2d2e4d2024934a35bff

**Comment by open-social-tugboat** [comment]
[Tugboat](https://www.tugboatqa.com) has finished building the preview for this pull request!

Link:
- https://pr4134-0tzpv5k2we7enrbv7adlld0e2udh7o0i.tugboatqa.com


Dashboard:
- https://dashboard.tugboatqa.com/6717d2d2e4d2024934a35bff

**Comment by open-social-tugboat** [comment]
[Tugboat](https://www.tugboatqa.com) has finished building the preview for this pull request!

Link:
- https://pr4134-0tzpv5k2we7enrbv7adlld0e2udh7o0i.tugboatqa.com


Dashboard:
- https://dashboard.tugboatqa.com/6717d2d2e4d2024934a35bff

**Comment by open-social-tugboat** [comment]
[Tugboat](https://www.tugboatqa.com) has finished building the preview for this pull request!

Link:
- https://pr4134-0tzpv5k2we7enrbv7adlld0e2udh7o0i.tugboatqa.com


Dashboard:
- https://dashboard.tugboatqa.com/6717d2d2e4d2024934a35bff

**Pull request merged**